File: mega-zep-frontend/src/main/angular/frontend/src/app/models/MonthlyReport/TimeWarning.ts

export class TimeWarning {

  private _date: Date;
  private _missingRestTime: number;
  private _missingBreakTime: number;
  private _excessWorkTime: number;
  // FIXME GAJ: what is stored in this array?!?
  private _warnings: Array<String>;

  get date(): Date {
    return this._date;
  }

  set date(value: Date) {
    this._date = value;
  }

  get missingRestTime(): number {
    return this._missingRestTime;
  }

  set missingRestTime(value: number) {
    this._missingRestTime = value;
  }

  get missingBreakTime(): number {
    return this._missingBreakTime;
  }

  set missingBreakTime(value: number) {
    this._missingBreakTime = value;
  }

  get excessWorkTime(): number {
    return this._excessWorkTime;
  }

  set excessWorkTime(value: number) {
    this._excessWorkTime = value;
  }

  get warnings(): Array<String> {
    return this._warnings;
  }

  set warnings(value: Array<String>) {
    this._warnings = value;
  }
}

File: mega-zep-backend/src/main/java/com/gepardec/mega/rest/model/TimeWarning.java
package com.gepardec.mega.rest.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;

import java.time.LocalDate;

public class TimeWarning {

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    private LocalDate date;
    private Double missingRestTime;
    private Double missingBreakTime;
    private Double excessWorkTime;

    public TimeWarning() {
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public Double getMissingRestTime() {
        return missingRestTime;
    }

    public void setMissingRestTime(Double missingRestTime) {
        this.missingRestTime = missingRestTime;
    }

    public Double getMissingBreakTime() {
        return missingBreakTime;
    }

    public void setMissingBreakTime(Double missingBreakTime) {
        this.missingBreakTime = missingBreakTime;
    }

    public Double getExcessWorkTime() {
        return excessWorkTime;
    }

    public void setExcessWorkTime(Double excessWorkTime) {
        this.excessWorkTime = excessWorkTime;
    }
}
